bitcart/sync.py

```diff
@@ -3,20 +3,34 @@
 import functools
 import inspect
 import threading
-from typing import Any, AsyncGenerator, Callable, Coroutine, List, Union
+from typing import Any, AsyncGenerator, Callable, List
 
 from .coins import COINS
 from .manager import APIManager
 from .providers.jsonrpcrequests import RPCProxy
 
 
+async def consume_generator(coroutine: AsyncGenerator) -> List[Any]:
+    return [i async for i in coroutine]
```

Comment:
try to learn this [i async for i in coroutine]. how does "async" work?

Reply:
It is not `async` construct, it is
`async for` construct.
It calls `__aiter__` method instead of `__iter__`
Basically, async version of for loop, useful for iterating over:
- Async generators
- Results fetched asynchronously from a network resource

Reply:
Thanks for your great explanation.